### PR TITLE
WW-89: api pagination for users

### PR DESCRIPTION
**Pull request by Noro911**
@th4tkh13m

### CODE REVIEW

File: src/main/java/com/example/codeE/controller/UserController.java [hunk 1]
```diff
@@ -17,9 +22,25 @@ public class UserController {
...
     @GetMapping
     @RequestMapping(value = "",method = RequestMethod.GET)
-    public ResponseEntity<?> getAllUsers(){
-        List<User> listaPersona = this.userImplement.getAllUsers();
-        return ResponseEntity.ok(listaPersona);
+    public ResponseEntity<?> getAllUsers(
+            @RequestParam(defaultValue = "") String role,
+            @RequestParam(defaultValue = "") String searchKeyword,
+            @RequestParam(defaultValue = "0") Integer pageNumber,
+            @RequestParam(defaultValue = "10") Integer pageSize
```
Comment (th4tkh13m) [inline src/main/java/com/example/codeE/controller/UserController.java:29]: @Noro911, can we do some validation here?

Please check this: https://www.baeldung.com/spring-validate-requestparam-pathvariable

The default value of pageNumber should be 1 instead of 0.

File: src/main/java/com/example/codeE/controller/UserController.java [hunk 2]
```diff
@@ -17,9 +22,25 @@ public class UserController {
 
     @GetMapping
     @RequestMapping(value = "",method = RequestMethod.GET)
-    public ResponseEntity<?> getAllUsers(){
-        List<User> listaPersona = this.userImplement.getAllUsers();
-        return ResponseEntity.ok(listaPersona);
+    public ResponseEntity<?> getAllUsers(
+            @RequestParam(defaultValue = "") String role,
+            @RequestParam(defaultValue = "") String searchKeyword,
```
Comment (th4tkh13m) [inline src/main/java/com/example/codeE/controller/UserController.java:27]: What's the point of setting these values ""?

File: src/test/java/com/example/codeE/controller/UserControllerTest.java
Comment (th4tkh13m) [inline src/test/java/com/example/codeE/controller/UserControllerTest.java:1]: More tests are desired.
Of course, you want to have tricky, edge cases instead of having one single test. Maybe we bound all of the constraints using the front end, but it should be helpful to create all those tricky test cases, such as having negative integers, extremely long queries, or blank queries.

**Review by th4tkh13m — CHANGES_REQUESTED** [review]

**Comment by Noro911** [comment]
@th4tkh13m 
Validation for User model -> done
Validation for Request param (API GET /users) -> done
Test coverage for getAllUsers with pagination api (controller) -> done
Test coverage for error response (controller) -> pending
Test coverage for all service methods related to API GET /users -> done

**Comment by th4tkh13m** [comment]
We merge this since it will help others in development.
But, I will open another PR related to this.
